automation/tests-storage.sh

```diff
@@ -12,28 +24,50 @@ create_loop_devices() {
 }
 
 setup_storage() {
-    python3 tests/storage/userstorage.py setup
+    # Configure lvm to ignore udev events, otherwise some lvm tests hang.
+    mkdir -p /etc/lvm
+    cp docker/lvmlocal.conf /etc/lvm/
+
+    # Make sure we have enough loop device nodes. Using 16 devices since with 8
+    # devices we have random mount failures.
+    create_loop_devices 16
+
+    # Build vdsm.
+    ./autogen.sh --system
+    make
+
+    # Setup user storage during the tests.
+    make storage
 }
 
 teardown_storage() {
-    python3 tests/storage/userstorage.py teardown \
+    make clean-storage \
         || echo "WARNING: Ingoring error while tearing down user storage"
 }
 
-# Configure lvm to ignore udev events, otherwise some lvm tests hang.
-mkdir -p /etc/lvm
-cp docker/lvmlocal.conf /etc/lvm/
-
-# Make sure we have enough loop device nodes. Using 16 devices since with 8
-# devices we have random mount failures.
-create_loop_devices 16
+# Process user argument
+user=$1
+if [ -z "$user" ]; then
+    echo "ERROR: user required"
+    print_help
+    exit 1
+fi
 
-# Build vdsm.
-./autogen.sh --system
-make
+echo "Running tests as user $user"
 
-# Setup user stoage during the tests.
-trap teardown_storage EXIT
 setup_storage
+if [ "$user" != "root" ]; then
+    # Change ownership of current and storage folders
+    # to allow non-privileged access.
+    chown -R vdsm:kvm ./ /var/tmp/vdsm*
```

Comment:
This hard codes "vdsm:kvm", but the user gave us the name of the user, so why not:

   chown -R %user:$user ...

This means that for running the tests as youself you will call:

    tests-storage.sh $USER

Reply:
Right, we should really use vdsm:kvm. But I agree that using vdsm or current user should be good enough for the tests, and we can always enhance this to use vdsm:kvm, and let the user pass both user and group. Maybe if the user does not pass a group we can default to the same value (correct for current user). The CI can pass vdsm:kvm to get more real environment.